src/main/room/main/G1HRPS/RoomManager.java

```diff
@@ -1,97 +1,305 @@
+ */
+public Room SearchList(String search_text) {  
+	for(var i=0; i<room_list_.size();i++) { //searching for same Room Name
+		if (room_list_.get(i).getRoomNumber()==search_text) {
+			return room_list.get(i);
+		}
 	}
+	for(var i=0; i<room_list_.size();i++) {
+		for(var j=0; j<room_list_.get(i).getGuestList().size();j++) { //searching for same Guest Name
+			if(room_list_.get(i).getGuestList().get(j).getName==search_text) {
+				return room_list.get(i);
+			}
+		}	
+	system.out.println("No room with details entered was found");
+	}	
+	throw new UnsupportedOperationException();
+}
+/**
+ * Removes a room from room_list_, user's input is the room number
+ * @param room_num_ Room number that user wants to remove from List.
+ */
+public void RemoveFromList(String roomNumber) { // it will not work if we just make a new room with the same parameters, since it will be stored in diff address. Suggest taking in roomNumber
+	Room room = SearchList(roomNumber);
+	system.out.println("Room with same room number found");
+	room_list_.remove(room);
+}
 
-	/**
-	 * Takes in an class object and list to add the object into.
-	 */
-	public void AddToList(Room room) {
-		// TODO - implement RoomManager.AddToList
-		throw new UnsupportedOperationException();
-	}
+/**
+ * Get room details of a room number
+ * @param roomNumber  This is the room number entered by user.
+ */
+public void GetRoomDetailByRoomNumber(String roomNumber) {
+	Room room = SearchList(roomNumber);
+	system.out.println("Room with same room number found");
+	room.viewRoomDetails;
+		}
+	}	
+}
 
-	/**
-	 * Takes in an class object and list to remove the object from the given list.
-	 */
-	public void RemoveFromList(Room room) {
-		// TODO - implement RoomManager.RemoveFromList
-		throw new UnsupportedOperationException();
-	}
+//does not deal with same names(rare case but still possible, it will print two room's details in this case)
+// Possible workaround by adding passport/driving license, but it will cause every room detail check to require to need to enter passport which is inefficient.
+/**
+ * Get room details by entering name of guest staying in that room
+ * @param guestName This is the name of the guest.
+ */
+public void GetRoomDetailByGuestName(String guestName) {
+	Room room = SearchList(guestName);
+	system.out.println("Room with same guest name found");
+	room.viewRoomDetails;
 
-	public void SearchList(String search_text) {
-		// TODO - implement RoomManager.SearchList
-		throw new UnsupportedOperationException();
-	}
+/**
+ * Checks whether a room is vacant or not.(currently)
+ * @param roomNumber Room number of room to be checked.
+ */
+public void CheckRoomAvailabilityByRoomNumber(String roomNumber) {// consider using room type to get max number of ppl staying in the room, and compare that to guest list of room
+		for(var i=0; i<room_list_.size();i++) {
+			if (room_list_.get(i).getRoomNumber()==roomNumber) {
+				room_list_.get(i).checkRoomAvailability();;
```

Comment:
I thought can just return room status?

Reply:
@Tlhoo this function needs to return something so that AppManager may use the result